core/sile.lua

```diff
@@ -240,10 +241,251 @@ function SILE.resolveFile(filename, pathprefix)
   return nil
 end
 
+SILE.currentCommandStack = {
+  --- Internal: push-pop balance checking ID
+  _lastPushId = 0,
+  --- Internal: Function assigned to frames to convert them to string
+  _frameToString = function(self, skipFile, withAttrs)
```

Comment:
This should probably be added to the frame prototype as a `__tostring()` function just like our other internal objects have dump functions.

Reply:
I had implemented it that way at first, but I couldn't find in documentation whether `__tostring` supports additional parameters. (And even if some implementation does, unless it is documented it could just be a bug.)

Reply:
It looks to me like `skipFile` is _always_ `false` and `withAttrs` is _always_ `true`, I can't see a code path where they are set differently. It's very difficult to evaluate what something should be called if there is no context for when it might be used. In this case I think I'd drop the arguments and just make that those the only behavior until such a time as the potential use for alternatives is clear.

In reviewing this, I also discovered the frame prototype _already has_ a `toString()` function from @simoncozens.

https://github.com/simoncozens/sile/blob/master/core/frame.lua#L74-L82

This should get re-used and extended if necessary, but I'd still argue we should make it a `__tostring()` meta property so it doesn't even have to be explicitly called (as we do for debugging other internal types).

Reply:
The parameters are used:
- `skipFile` [here](https://github.com/simoncozens/sile/pull/603/files#diff-8e3ebbe41b16928909ff53e84982f1d1R430) and a few more other places
- `withAttrs` is usually unspecified and thus `false` (`nil`)

So it would be incorrect to drop them.

That `frame.lua` refers to a content frame. What I refer to as a frame here is a [stack trace frame](https://en.wikipedia.org/wiki/Call_stack#STACK-FRAME).